crates/uv-resolver/src/pubgrub/dependencies.rs

```diff
@@ -23,23 +23,26 @@ pub(crate) struct PubGrubDependency {
     /// even if this field is None where there is an override with a URL or there is a different
     /// requirement or constraint for the same package that has a URL.
     pub(crate) url: Option<VerbatimParsedUrl>,
+    /// The local version for this requirement, if specified.
+    pub(crate) local: Option<Version>,
```

Comment:
Is it true that if `local` is `Some`, then `Version::is_local` will always be true?

If so, then perhaps this could just be a `Version`, and case analysis uses `Version::is_local` or `Version::local`.

Reply:
After reading through the PR, I retract my suggestion. I see how this is very convenient because the acquisition of a local-only `Version` is pretty naturally expressed via `Option<Version>`. And I don't think you ever look at a `local` and try to access its local segments, so there isn't really any redundant case analysis.

Reply:
> Is it true that if local is Some, then Version::is_local will always be true?

This is actually not true, because we also extract local versions from direct URL/path requirements.